src/pkgcore/config/hint.py

```diff
@@ -3,10 +3,21 @@
 
 __all__ = ("ConfigHint", "configurable")
 
+import typing
+
 
 class ConfigHint:
     """Hint for introspection supplying overrides."""
 
+    types: dict[str, str]
+    positional: tuple[str, ...]
+    required: tuple[str, ...]
+    typename: typing.Optional[str]
+    allow_unknowns: bool
+    authorative: bool
+    requires_config: bool
+    raw_class: bool
```

Comment:
It's best to assume I'm new to python typing, thus best practices I'm still translating from other languages.  As to this- I was using `mypy --strict` and just iterating through the config system until I felt like I had a decent handle on the typing and how it was working, thus me enumerating a bunch of stuff that perhaps isn't needed.  I assume our end goal is to make typing mandatory (IE, failed build if the typing is unclear), thus enumerating this.

So... two asks:
1) what should this actually be in your view?
2) If either of y'all have a codebase (or two) that you consider to be best practices for typing, I'd appreciate if you could point me at it so I can dig through it and internalize it.

Reply:
@arthurzam I appreciate the merge, but this question still stands- if you've got pointers for what I should be emulating (code style wise), let me know.

Hell, if there are examples in pkgcore codebase, just point me there and I'll integrate it.

Reply:
@ferringb Let me say something weird - I've no idea. Let's build a "new" standard that looks the best by your eyes, and with time we will learn. If it would make you laugh, just know that my current python style is based on nattka/pkgcheck/pkgdev and sometimes pkgcore, so I'm somewhat of your style follower.

Just go with what you can type. The syntax is less important for me. My end goal is that mypy or an IDE (I use vscode) can parse the types and show me when developing, which would make everyone's life easier. If you want inline, class level, even `*.pyi` - I don't care as long as we move to better future. Nothing becomes perfect immediately, and we improve things as we go.

After currently release of pkgcore, pkgcore drop <3.10 python support, so we would be able to use 3.10 syntax finally. I've no issues with any syntax (match-case, `|` for type union, ...) so we would be able to modernize more the code.